### ISSUE TITLE
undefined method `for_all_cops'

### ISSUE THREAD

**Issue by dankohn** [bbatsov/rubocop#2965]
Rubocop master (commit dce26dec8) is producing an error when running with https://github.com/nevir/rubocop-rspec .
## Expected behavior

Rubocop runs

## Actual behavior
```bash
$ rubocop
undefined method `for_all_cops' for #<Hash:0x007fade4451930>
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/config_loader.rb:120:in `merge_with_default'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-rspec-1.4.0/lib/rubocop/rspec/inject.rb:17:in `defaults!'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/rubocop-rspec-1.4.0/lib/rubocop-rspec.rb:9:in `<top (required)>'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/config_loader_resolver.rb:16:in `require'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/config_loader_resolver.rb:16:in `block in resolve_requires'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/config_loader_resolver.rb:12:in `each'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/config_loader_resolver.rb:12:in `resolve_requires'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/config_loader.rb:39:in `load_file'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/config_loader.rb:97:in `configuration_from_file'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/config_store.rb:41:in `for'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/target_finder.rb:113:in `excluded_dirs'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/target_finder.rb:91:in `find_files'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/target_finder.rb:61:in `target_files_in_dir'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/target_finder.rb:32:in `find'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/runner.rb:47:in `find_target_files'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/runner.rb:31:in `run'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/lib/rubocop/cli.rb:30:in `run'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/bin/rubocop:14:in `block in <top (required)>'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/2.2.0/benchmark.rb:303:in `realtime'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bundler/gems/rubocop-dce26dec8a7b/bin/rubocop:13:in `<top (required)>'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bin/rubocop:23:in `load'
/Users/dan/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/bin/rubocop:23:in `<main>'
```
## Steps to reproduce the problem

Error occurs on dce26dec8. Version 5a4c302 works fine. Commenting out `require: rubocop-rspec` causes dce26dec8 to work correctly.

## RuboCop version

Note this is the same on both commits:
```
$ rubocop -V
0.38.0 (using Parser 2.3.0.6, running on ruby 2.2.3 x86_64-darwin15)
```

.rubocop.yml is:
```yml
require: rubocop-rspec
Rails:
  Enabled: true
AllCops:
  TargetRubyVersion: 2.2
  DisplayCopNames: true
  DisplayStyleGuide: true
  Include:
    - '**/*.pryrc'
    - '**/*.rake'
    - '.simplecov'
    - '**/config.ru'
    - '**/Gemfile'
    - '**/Guardfile'
    - '**/Rakefile'
    - '**/Vagrantfile'
  Exclude:
    - 'db/schema.rb'
    - 'features/support/_circleci_formatter.rb'
    - 'lib/generators/**/*'
    - 'vendor/**/*'
Documentation:
  Enabled: false
Lint/AmbiguousRegexpLiteral:
  Exclude:
    - 'features/step_definitions/*'
Lint/ImplicitStringConcatenation:
  Enabled: false  
Metrics/AbcSize:
  Exclude:
    - 'db/migrate/*'
Metrics/ClassLength:
  Exclude:
    - 'db/migrate/*'
Metrics/LineLength:
  Exclude:
    - 'db/migrate/*'
Metrics/MethodLength:
  Exclude:
    - 'db/migrate/*'
RSpec/AnyInstance:
  Enabled: false
RSpec/DescribeClass:
  Exclude:
    - 'spec/requests/**/*'
RSpec/FilePath:
  Exclude:
    - 'spec/routing/**/*'
Style/ArrayJoin:
  Enabled: false
Style/ClassAndModuleChildren:
  Enabled: false
Style/CollectionMethods:
  Enabled: true
Style/FileName:
  Exclude:
    - 'bin/*'
Style/NestedParenthesizedCalls:
  Exclude:
    - 'features/step_definitions/*'
    - 'spec/**/*'
Style/TrivialAccessors:
  Enabled: true
  ExactNameMatch: true
```

**Comment by aried3r** [comment]
Facing the same problem using RuboCop 0.39.0
```Shell
$ rubocop -V
0.39.0 (using Parser 2.3.0.7, running on ruby 2.3.0 x86_64-darwin15)
```

Is this a problem with rubocop or rubocop-rspec?

**Comment by bbatsov** [comment]
Likely it's a problem with `rubocop-rspec`, as this is definitely not affecting RuboCop itself.

**Comment by TommyByrne** [comment]
@bbatsov this commit https://github.com/td-berlin/td_critic/pull/25 seems to point to a change within Rubocop itself that might be affecting things. Not sure what the next best step is, just wanted to put that on your radar. Thanks for Rubocop!!

**Comment by nijikon** [comment]
This is a rubocop-rspec issue, fixed in nevir/rubocop-rspec#80. @bbatsov would it be possible to extend ConfigLoader.load_file with extra validate parameter? In that way I could use this method to load rubocop-rspec file. Need this flag cause when it validates I get warnings :) Or maybe I'm not doing it correctly.

**Comment by dankohn** [comment]
Fixed with https://github.com/nevir/rubocop-rspec/commit/7b02959ac37fb3e0d6aaa9a782d3c45054d81c27
